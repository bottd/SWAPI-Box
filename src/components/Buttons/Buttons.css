.Buttons {
  display: flex;
  justify-content: space-evenly;
}

button {
  background-color: transparent;
  border: 4px solid #fdde09;
  border-radius: 30px;
  color: #fdde09;
  cursor: pointer;
  font-family: starWars;
  font-size: 28px;
  height: 60px;
  outline: 0;
  transition: all 0.4s ease-in-out;
  width: 180px;
}

button:hover {
  background-color: #fdde09;
  border: 4px solid black;
  color: black;
}
