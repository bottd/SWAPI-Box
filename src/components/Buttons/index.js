import React from "react";
import PropTypes from "prop-types";
import "./Buttons.css";
import * as API from "../../helper/helper";

const Buttons = ({ updateData }) => {
  return (
    <div className="button-section">
      <button
        onClick={() => API.getPeople()
          .then(people => updateData('people',people))}
        >people</button>
      <button>planets</button>
      <button onClick={() => API.getVehicles()
        .then(vehicles => updateData('vehicles',vehicles))
      }>vehicles</button>
    </div>
  );
};

Buttons.propTypes = {
  updateData: PropTypes.func.isRequired
};

export default Buttons;
