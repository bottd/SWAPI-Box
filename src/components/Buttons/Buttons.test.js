import React from "react";
import Buttons from "./index";
import { shallow } from "enzyme";

describe("Buttons", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Buttons />);
  });

  it("", () => {

  });
  
});
