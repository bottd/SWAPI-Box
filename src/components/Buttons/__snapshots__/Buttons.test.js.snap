// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Rendering Renders like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Buttons
    selectedCategory="planets"
    updateData={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="people"
          onClick={[Function]}
          to="/people"
        >
          people
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="planets"
          onClick={[Function]}
          to="/planets"
        >
          planets
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="vehicles"
          onClick={[Function]}
          to="/vehicles"
        >
          vehicles
        </NavLink>,
      ],
      "className": "Buttons",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "people",
          "className": "people",
          "onClick": [Function],
          "to": "/people",
        },
        "ref": null,
        "rendered": "people",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "planets",
          "className": "planets",
          "onClick": [Function],
          "to": "/planets",
        },
        "ref": null,
        "rendered": "planets",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "vehicles",
          "className": "vehicles",
          "onClick": [Function],
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": "vehicles",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="people"
            onClick={[Function]}
            to="/people"
          >
            people
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="planets"
            onClick={[Function]}
            to="/planets"
          >
            planets
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="vehicles"
            onClick={[Function]}
            to="/vehicles"
          >
            vehicles
          </NavLink>,
        ],
        "className": "Buttons",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "people",
            "className": "people",
            "onClick": [Function],
            "to": "/people",
          },
          "ref": null,
          "rendered": "people",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "planets",
            "className": "planets",
            "onClick": [Function],
            "to": "/planets",
          },
          "ref": null,
          "rendered": "planets",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "vehicles",
            "className": "vehicles",
            "onClick": [Function],
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "vehicles",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
